src/QueryLanguage/Walker/PhpCr/NodeWalker.php

```diff
@@ -0,0 +1,169 @@
+use Fazland\ApiPlatformBundle\QueryLanguage\Expression\ExpressionInterface;
+use Fazland\ApiPlatformBundle\QueryLanguage\Expression\Literal\LiteralExpression;
+use Fazland\ApiPlatformBundle\QueryLanguage\Expression\Literal\NullExpression;
+use Fazland\ApiPlatformBundle\QueryLanguage\Expression\ValueExpression;
+use Fazland\ApiPlatformBundle\QueryLanguage\Walker\AbstractWalker;
+use PHPCR\Query\QOM\QueryObjectModelConstantsInterface as QOMConstants;
+
+class NodeWalker extends AbstractWalker
+{
+    private const COMPARISON_MAP = [
+        '=' => QOMConstants::JCR_OPERATOR_EQUAL_TO,
+        '<' => QOMConstants::JCR_OPERATOR_LESS_THAN,
+        '<=' => QOMConstants::JCR_OPERATOR_LESS_THAN_OR_EQUAL_TO,
+        '>' => QOMConstants::JCR_OPERATOR_GREATER_THAN,
+        '>=' => QOMConstants::JCR_OPERATOR_GREATER_THAN_OR_EQUAL_TO,
+        'like' => QOMConstants::JCR_OPERATOR_LIKE,
+    ];
+
+    /**
+     * @var string|null
+     */
+    private $fieldType;
+
+    public function __construct(string $field, ?string $fieldType = null)
+    {
+        parent::__construct($field);
+
+        $this->fieldType = $fieldType;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function walkLiteral(LiteralExpression $expression)
+    {
+        $value = parent::walkLiteral($expression);
+        if ($expression instanceof NullExpression) {
+            return $value;
+        }
+
+        if ('date' === $this->fieldType) {
+            return new \DateTime($value);
+        }
+
+        return $value;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function walkComparison(string $operator, ValueExpression $expression)
+    {
+        $comparison = new ConstraintComparison(new DummyNode(), self::COMPARISON_MAP[$operator]);
+
+        $field = $this->field;
+        $left = function (AbstractNode $parent) use ($field): AbstractNode {
+            return 'nodename' === $this->fieldType ?
+                new OperandDynamicLocalName($parent, \explode('.', $field)[0]) :
+                new OperandDynamicField($parent, $field);
+        };
+
+        if ('like' === $operator) {
+            $lowerCase = new OperandDynamicLowerCase($comparison);
+            $lowerCase->addChild($left($lowerCase));
+            $left = $lowerCase;
+
+            $value = '%'.$expression->getValue().'%';
+        } else {
+            $left = $left($comparison);
+            $value = $expression->getValue();
+        }
+
+        $right = new OperandStaticLiteral($comparison, $value);
+
+        $comparison->addChild($left);
+        $comparison->addChild($right);
+
+        return $comparison;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function walkAll()
+    {
+        // Do nothing.
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function walkOrder(string $field, string $direction)
+    {
+        $left = function (AbstractNode $parent) use ($field): AbstractNode {
+            return 'nodename' === $this->fieldType ?
```

Comment:
constants?

Reply:
As all field types, `nodename` has no constant in phpcr odm library

Reply:
👍